graphql_compiler/compiler/expressions.py

```diff
@@ -894,12 +895,18 @@ def to_match(self):
         negated_regular_operator_format = '(NOT (%(left)s %(operator)s %(right)s))'
         # pylint: enable=unused-variable
 
+        # Comparing null to a value does not make sense.
+        if self.left == NullLiteral:
+            raise AssertionError(u'The left expression cannot be a NullLiteral! Received operator '
+                                 u'{} and right expression {}.'.format(self.operator, self.right))
```

Comment:
There's a lot of checks and complexity here. Would it be easier to use a `UnaryTransformation` for `is_null` and `is_not_null`? They are after all unary `is_null : Any -> Bool`.

Reply:
They are, but I think we'd get more value from using the same representation as the other null checks. For example, the null checks generated from lowering `ContextFieldExistence` expressions all use `BinaryComposition` already, and I don't think there's a lot of value in refactoring all that code to use unary transforms instead.

Reply:
I'm not convinced. In person?

Reply:
Resolved offline.